### ISSUE TITLE
How to get yaw/pitch/roll data from object?

### ISSUE THREAD

**Issue by undera** [ccappelle/pyrosim#3]
Hi,

I look for a way to get orientation data from simulation, like position sensor, but with rotation values. I expected vestibular sensor to offer this, but it seems vestibular only gives single value (and not well explained nature of this value in docs or code docs).

How can I get rotation of object? If there is no such sensor, would it be possible to add one?

Thanks for your project!

**Comment by undera** [comment]
Btw, I made monkey patching to Vestibular sensor so I got the rotation values from it. Looks like pyrosim needs completely new sensor class. Unfortunately, I'm not good at C + Python connection here, to provide PR... I can only show my "hacked" source if it helps.

**Comment by ccappelle** [comment]
I was not the one to create the original vestibular sensor so I have left it alone. I have been meaning to create more of a class hierarchical structure which should make it easier to add new sensors. You should continue to use your vestibular sensor because I won't be changing it in the main project.

**Comment by undera** [comment]
This is probably not the best way. The point of Open Source is to improve from community feedback.

**Comment by ccappelle** [comment]
I agree, but this is intended for use in a very specific setting, i.e. a college course. The whole point is for it to be as simple as possible to use for people who have never used simulators or even programmed before. I am also trying to keep it as similar as to previous versions to keep experiments which are currently running as similar as possible. So changing the way vestibular sensors work would ruin other peoples current experiments.

**Comment by undera** [comment]
My proposition is to add new sensor type, thus no existing scripts will be affected. Rotational sensor is quite generic thing, I believe it is useful for pyrosim to have this capability.

**Comment by undera** [comment]
I have made a PR to implement it : https://github.com/ccappelle/pyrosim/pull/5
